firmware/src/can_app.c

```diff
@@ -1,16 +1,31 @@
 #include "can_app.h"
 
+uint32_t can_app_send_state_clk_div;
+uint32_t can_app_send_tachometer_clk_div;
+
+extern tachometer_t tachometer;
+extern state_machine_t state_machine;
+extern error_flags_t error_flags;
+
 /**
  * @brief Prints a can message via usart
  */
 
 void compute_rpm_avg(void)
 {
-    if(tachometer.rpm_avg_sum_count){
-        tachometer.rpm_avg = tachometer.rpm_avg_sum/tachometer.rpm_avg_sum_count;
-        tachometer.rpm_avg_sum = 0;
-        tachometer.rpm_avg_sum_count = 0;
+    if(tachometer.dt_avg_sum_count){
+		tachometer.lock = 1;
+        tachometer.dt_avg = tachometer.dt_avg_sum/tachometer.dt_avg_sum_count;
     }
+    tachometer.dt_avg_sum = 0;
+    tachometer.dt_avg_sum_count = 0;
+	tachometer.lock = 0;
+	tachometer.rpm = 60 * 1e6 /  (tachometer.dt_avg * 64);
```

Comment:
```suggestion
    static uint8_t zeroes_counter = 0;
    c = 1;
    if(tachometer.dt_avg_sum_count && tachometer.dt_avg_sum){
        tachometer.dt_avg = tachometer.dt_avg_sum/tachometer.dt_avg_sum_count;
        tachometer.dt_avg_sum = 0;
        tachometer.dt_avg_sum_count = 0;
	      
	      tachometer.rpm = 60 * 1e6 / (tachometer.dt_avg * 64);
	      zeroes_counter = 0;
	  }else{
	      if(++zeroes_counter == 5) tachometer.rpm = 0;
	  }
	  tachometer.lock = 0;
```

reasoning:
- set RPM to zero after 5 average computation tries with incomplete turns.

note:
- tachometer.lock needs to be set before any read to tachometer variables